pennylane/transforms/optimization/cancel_inverses.py

```diff
@@ -165,9 +168,8 @@ def qfunc(x, y, z):
         # - there is no wire symmetry
         # - the control wire symmetry does not apply because the control wires are not the same
         # - neither of the flags are_self_inverses and are_inverses are true
-        apply(current_gate)
+        operations.append(current_gate)
         continue
 
-    # Queue the measurements normally
-    for m in tape.measurements:
-        apply(m)
+    new_tape = QuantumTape(operations, tape.measurements, shots=tape.shots)
```

Comment:
```suggestion
    new_tape = QuantumScript(operations, tape.measurements, shots=tape.shots)
```

Reply:
We might also need to copy over `tape._qfunc_output` and `tape.trainable_params`.